src/sharect.js

```diff
@@ -116,16 +138,28 @@ const Sharect = (function(){
 
       div.appendChild(arrow)
 
-      document.body.appendChild(div)
+      if (hasSelection()) {
+        offsetParent(_selection.anchorNode.parentElement).appendChild(div)
+      } else {
+        document.body.appendChild(div)
+      }
     }
 
     function attachEvents() {
-      function hasSelection() {
-        return !!window.getSelection().toString()
-      }
 
       function hasTooltipDrawn() {
```

Comment:
Avoid side effects.
This function should return just `true` or `false`. If DOM manipulation or any other change is needed, it should be done outside of this function.
The `mouseup` event already tests for tooltip drawn and if it has selection. Maybe you could add another test there to look at its `offsetParent`.